scrapers/errata.py

```diff
@@ -68,7 +69,55 @@ def update_neo4j(self, advisories):
                 'type_': advisory['type'],
                 'update_date': advisory['update_date'],
                 'updated_at': advisory['updated_at']
-            })[0]
+            }
+
+            associated_builds = self.get_associated_builds(advisory['id'])
+            if not associated_builds:
+                adv = Advisory.create_or_update(adv_params)[0]
+
+            for associated_build in associated_builds:
+                build = ContainerKojiBuild.nodes.get_or_none(id_=associated_build['id_'])
+                if not build:
+                    build = KojiBuild.nodes.get_or_none(id_=associated_build['id_'])
+
+                container_adv = False
+                if build:
+                    if build.__label__ == 'ContainerKojiBuild':
+                        adv = self.create_container_build_advisory(adv_params)
+                        container_adv = True
+                    else:
+                        adv = Advisory.create_or_update(adv_params)[0]
+
+                # If this is set, that means it was once part of the advisory but not anymore.
+                # This relationship needs to be deleted if it exists.
+                if associated_build['removed_index_id']:
+                    if build:
+                        adv.attached_builds.disconnect(build)
+                else:
+                    # This key shouldn't be stored in Neo4j
+                    del associated_build['removed_index_id']
+                    # Query Teiid only if the build is not present in Neo4j
+                    if build:
+                        if container_adv:
+                            is_container_build = True
+                        else:
+                            is_container_build = False
+                    else:
+                        attached_build = self.get_koji_build(associated_build['id_'])
+                        if attached_build:
+                            is_container_build = self.is_container_build(attached_build)
+                        else:
+                            is_container_build = False
+
+                    if is_container_build:
+                        build = ContainerKojiBuild.get_or_create(
+                            {'id_': associated_build['id_']})[0]
+                        adv = self.create_container_build_advisory(adv_params)
```

Comment:
This gets called in every for loop so it seems like unnecessary churn on the Neo4j server. I have an idea around this. I'll swing by in a few mins to discuss.